skimage/filters/_diffusion_utils.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+# from numba import jit
+
+
+# pythran export nonlinear_aniso_step(float64[:,:], float64[:,:], float64[:,:], float64[:,:],float64[:,:], float or int, uint8)
+def nonlinear_aniso_step(src, dest, a, b, c, tau, border):
+    for i in range(border, src.shape[0] - border - 1):
+        for j in range(border, src.shape[1] - border - 1):
+            dest[i, j] = src[i, j] + tau * (src[i - 1, j + 1] *
+                ((abs(b[i - 1, j + 1]) - b[i - 1, j + 1]) / 4.0 + (abs(b[i, j]) - b[i, j]) / 4.0) + src[i, j + 1] *
```

Comment:
Looking at this, I wonder if it cannot be written as an array operation?

Reply:
I don't know if it's possible. I cannot see how. 
This is the formula from this [Weickert book](https://www.mia.uni-saarland.de/weickert/Papers/book.pdf), page  95.